### PR TITLE
Feature/proj 69

### PR DESCRIPTION
**Pull request by Lelik7777**
# Feature: SignIn and AuthCard Enhancements

## Overview

This pull request introduces several enhancements and new features related to the SignIn and AuthCard components. It includes the creation of new folders with templates, updates to component logic, styles, and documentation improvements. Additionally, it involves updating dependencies to ensure compatibility and improved functionality.

### SignIn Component

- **Folder Structure**: Added a new `signIn` folder with relevant templates to organize the SignIn component files.
- **Layout and Styles**: Introduced a new layout and updated styles for the SignIn component.
  - Updated `_variables.scss` to include new style variables.
  - Updated `signIn.module.scss` for improved styling.
- **Component Updates**:
  - Updated `signIn.tsx` to enhance functionality and remove unused imports.
  - Added `ErrorMessage` and `FormButtons` components for better error handling and form actions.
  - Integrated `useSignIn` hook for managing form state and logic.
- **Dependency Updates**:
  - Installed `react-hook-form` for form management.
  - Updated `@photo-fiesta/ui-lib` to version 0.0.6.
  - Updated `@hookform/resolvers` to version 3.9.0.
  - Installed `zod` version 3.23.8 for schema validation.

### AuthCard Component

- **Folder Structure**: Added a new `authCard` folder with relevant templates to organize the AuthCard component files.
- **Logic Relocation**: Moved certain logic from the SignIn component to the AuthCard component for better modularity and reusability.
- **Component Updates**:
  - Updated `authCard.tsx` to include JSDoc comments for better documentation and understanding.
  - Updated `authCard.module.scss` for improved styling.

### Refactoring

- **Index File**: Updated `index.tsx` where the AuthCard component is rendered to reflect the changes and improvements.

## Documentation

- Added JSDoc comments to both the `AuthCard` and `SignIn` components, providing detailed descriptions and usage examples.

## Testing

- [ ] Ensure all components render correctly with the updated styles and logic.
- [ ] Verify that form submission and validation work as expected with the new `useSignIn` hook and `zod` validation.
- [ ] Test the integration of the updated dependencies to ensure compatibility.

## Screenshot
Screenshot_1

### CODE REVIEW

**Review by EvgeniyaYakunina — APPROVED** [review]

File: src/features/auth/ui/authCard/authCard.module.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  width: var(--google-link-size);
+  height: var(--google-link-size);
+}
+
+.footer {
+  display: flex;
+  flex-direction: column;
+  gap: 6px;
+  align-items: center;
+}
```
Comment (EvgeniyaYakunina) [inline src/features/auth/ui/authCard/authCard.module.scss:32]: Добавить адаптив для мобильных устройств

File: src/features/auth/ui/signIn/signIn.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ *   return <SignIn onSubmit={handleSubmit} />
+ * }
+ */
+export const SignIn = ({ onSubmit }: SignInProps) => {
+  const { control, errors, handleSubmit, isDirty, isValid, onSubmitForm } = useSignIn(onSubmit)
+  const classNames = {
+    container: styles.container,
+    form: styles.form,
+  } as const
+  const isButtonDisabled = !isValid || !isDirty
```
Comment (EvgeniyaYakunina) [inline src/features/auth/ui/signIn/signIn.tsx]: Проверить необходимость этой константы, так как по макету кнопка не дизейблится

File: src/features/auth/ui/signIn/signIn.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+  return (
+    <AuthCard
+      footerLinkHref={'/sign-up'}
```
Comment (EvgeniyaYakunina) [inline src/features/auth/ui/signIn/signIn.tsx:43]: нужна todo, так как линки нужно будет проверить после того как определимся с нэймингом путей.

File: src/features/auth/ui/signIn/signIn.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                label={'Password'}
+                placeholder={'Enter your password'}
+                variant={'password'}
+                {...field}
+              />
+            )}
+          />
+          {(errors.email || errors.password) && <ErrorMessage />}
+        </div>
+        <FormButtons isButtonDisabled={isButtonDisabled} />
```
Comment (EvgeniyaYakunina) [inline src/features/auth/ui/signIn/signIn.tsx]: А откуда у нас такая компонента FormButtons?  Везде используется просто кнопка сабмит....

File: src/features/auth/ui/signIn/useSignIn.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import { z } from 'zod'
+
+export type FormInputs = z.infer<typeof signInSchema>
+const signInSchema = z.object({
+  email: z.string().email('Invalid email address'),
+  password: z
+    .string()
+    .min(6, 'Password must be at least 8 characters')
+    .max(20, 'Password must be at most 32 characters')
+    .regex(/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*()_+\-=\\[\]{};':"\\|,.<>/?]).*$/),
```
Comment (EvgeniyaYakunina) [inline src/features/auth/ui/signIn/useSignIn.ts]: создала для regex константу в папке shared/config

File: src/features/auth/ui/signIn/signIn.tsx (hunk 2 above)
Comment (Lelik7777) [reply src/features/auth/ui/signIn/signIn.tsx]: исправил

File: src/features/auth/ui/signIn/signIn.tsx (hunk 3 above)
Comment (Lelik7777) [reply src/features/auth/ui/signIn/signIn.tsx:43]: исправил

File: src/features/auth/ui/signIn/signIn.tsx (hunk 4 above)
Comment (Lelik7777) [reply src/features/auth/ui/signIn/signIn.tsx]: это блок с кнопками, вынес его отдельно. В нем присутствует <Button variant={'primary'}>Sign in</Button>, которая и реализует submit

File: src/features/auth/ui/authCard/authCard.module.scss (hunk 1 above)
Comment (Lelik7777) [reply src/features/auth/ui/authCard/authCard.module.scss:32]: добавим через отдельную таску после мержа proj-82

File: src/features/auth/ui/signIn/useSignIn.ts (hunk 5 above)
Comment (Lelik7777) [reply src/features/auth/ui/signIn/useSignIn.ts]: добавим через отдельную таску после мержа proj-82

**Review by Emilo067 — APPROVED** [review]

**Review by EvgeniyaYakunina — APPROVED** [review]

**Pull request merged**